samza-sql/src/main/java/org/apache/samza/sql/translator/QueryTranslator.java

```diff
@@ -52,6 +65,27 @@
   private final ModifyTranslator modifyTranslator;
   private final SamzaSqlApplicationConfig sqlConfig;
   private final Map<String, SamzaRelConverter> converters;
+  private static final String LOG_OUTPUT_STREAM = "log.outputStream";
+
+  private static class OutputMapFunction implements MapFunction<SamzaSqlRelMessage, KV<Object, Object>> {
+    private transient SamzaRelConverter samzaMsgConverter;
+    private final String outputTopic;
+
+    OutputMapFunction(String outputTopic) {
+      this.outputTopic = outputTopic;
+    }
+
+    @Override
+    public void init(Config config, TaskContext taskContext) {
+      TranslatorContext context = (TranslatorContext) taskContext.getUserContext();
```

Comment:
Does this PR fix the "different TranslatorContext instance for different query statements" problem?

Reply:
No. PR https://github.com/apache/samza/pull/697 is for the issue we discussed last week "different TranslatorContext instance for different query statements".